warehouse/query-core/src/main/java/datawave/query/jexl/functions/FiAggregator.java

```diff
@@ -0,0 +1,202 @@
+package datawave.query.jexl.functions;
+
+import datawave.marking.ColumnVisibilityCache;
+import datawave.query.attributes.Attribute;
+import datawave.query.attributes.AttributeFactory;
+import datawave.query.attributes.Document;
+import datawave.query.attributes.DocumentKey;
+import datawave.query.data.parsers.FieldIndexKey;
+import datawave.query.data.parsers.KeyParser;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.predicate.EventDataQueryFilter;
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.hadoop.io.Text;
+
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * A field index aggregator for event queries
+ */
+public class FiAggregator implements FieldIndexAggregator{
+
+    private static final Text EMPTY_TEXT = new Text();
+
+
+    private int maxNextCount = -1;
+    protected Set<String> fieldsToKeep;
+    protected EventDataQueryFilter filter;
+
+
+    protected final FieldIndexKey parser = new FieldIndexKey();
+    protected final FieldIndexKey tkParser = new FieldIndexKey();
+
+
+    public FiAggregator() {
+        //  empty constructor
+    }
+
+    //  builder style methods for legacy values
```

Comment:
If these are 'builder style' shouldn't they return 'this' so calls to these methods can be chained?

Reply:
Updated in https://github.com/NationalSecurityAgency/datawave/pull/1971/commits/b596c4a93cecbe08befbda16e24c4af06dd74000